src/dmd/parse.d

```diff
@@ -5030,10 +5030,8 @@ final class Parser(AST) : Lexer
 
         const(char)* sp = !ident ? "" : " ";
         const(char)* s = !ident ? "" : ident.toChars();
-        if (alt & 1) // contains C-style function pointer syntax
-            error(loc, "instead of C-style syntax, use D-style `%s%s%s`", t.toChars(), sp, s);
-        else
-           dmd.errors.deprecation(loc, "instead of C-style syntax, use D-style syntax `%s%s%s`", t.toChars(), sp, s);
+        // @@@DEPRECATED_2.087_REMOVE@@@
```

Comment:
Do we plan to ever deprecate this?
It seems like a useful error message to keep around forever (it was very helpful for the C->D conversion).

Reply:
Actually, I hope so.  I don't think we should spend logic detecting invalid C syntax just so we can display an error.  I'm hoping we can just rely on ordinary parsing errors.  But that's just my opinion.  I'll go along with whatever the team decides.

Reply:
I used to think the same as you @JinShil then I ported some C and C++ code to D. If the overhead isn't too much (it's usually a few lines), it's definitely worth it.